packs/elastic/actions/lib/static_metagen.py

```diff
@@ -0,0 +1,122 @@
+import yaml
+
+class StaticMetagen(object):
+
+    action_meta = {
+        "name": "",
+        "parameters": {
+            "action": {
+                "type": "string",
+                "immutable": True,
+                "default": ""
+            },
+            "host": {
+                "description": "Elasticsearch host.",
+                "type": "string",
+                "required": True
+            },
+            "url_prefix": {
+                "description": "Elasticsearch http url prefix.",
+                "type": "string"
+            },
+            "port": {
+                "description": "Elasticsearch port.",
+                "type": "string",
+            },
+            "use_ssl": {
+                "description": "Connect to Elasticsearch through SSL.",
+                "type": "boolean",
+                "default": False
+            },
+            "http_auth": {
+                "description": "Use Basic Authentication ex: user:pass",
+                "type": "string"
+            },
+            "master_only": {
+                "description": "Only operate on elected master node.",
+                "type": "boolean",
+                "default": False
+            },
+            "timeout": {
+                "description": "Elasticsearch operation timeout in seconds.",
+                "type": "integer",
+                "default": 600
+            },
+            "log_level": {
+                "description": "Log level [critical|error|warning|info|debug].",
+                "type": "string",
+                "default": "warn"
+            },
+            "dry_run": {
+                "description": "Do not perform any changes.",
+                "type": "boolean",
+                "default": False
+            }
+        },
+        "runner_type": "run-python",
+        "description": "Run a Meta Action through a generic Runner.",
+        "enabled": True,
+        "entry_point": "curator_runner.py"}
+
+    parameter_meta = {
+        "type": "string"
+    }
+
+    def __init__(self, action_meta=None):
+        self.action_meta = StaticMetagen.action_meta
+        if action_meta is not None:
+            self.action_meta.update(action_meta)
+
+
+    def generate_action(self, module_type, action):
+        manifest = self.action_meta
+        manifest['name'] = "{0}_{1}".format(module_type, action)
+        manifest['parameters']['action']['default'] = action
+
+        fh = open('{0}_{1}.yaml'.format(module_type, action), 'w')
+        fh.write('---\n')
+        fh.write(yaml.dump(manifest, default_flow_style=False))
+        fh.close()
+
+
+    def generate_from_file(self, meta_file):
+        if meta_file is None:
+            return None
+        fh = open(meta_file)
```

Comment:
You never close the original file.

Using something like `with` context manager is better and safer since you don't need to manually make sure to close the file.